nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/ShellUserGroupProvider.java

```diff
@@ -339,6 +349,26 @@ private long getDelayProperty(AuthorizerConfigurationContext authContext, String
         return syncInterval;
     }
 
+    private long getTimeoutProperty(AuthorizerConfigurationContext authContext, String propertyName, String defaultValue) {
```

Comment:
I like that this logic is encapsulated in a private accessor method. Very clean design.